basin3d/plugins/epa.py

```diff
@@ -63,14 +66,61 @@
+# https://www.epa.gov/waterdata/water-quality-portal-quick-reference-guide
+# accessed 2024-09-23
+FIELD_NAMES = {
+    'loc_id': {'2.2': 'MonitoringLocationIdentifier', '3.0': 'Location_Identifier'},
+    'lat': {'2.2': 'LatitudeMeasure', '3.0': 'Location_Latitude'},
+    'long': {'2.2': 'LongitudeMeasure', '3.0': 'Location_Longitude'},
+    'org_name': {'2.2': 'OrganizationFormalName', '3.0': 'Org_FormalName'},
+    'loc_name': {'2.2': 'MonitoringLocationName', '3.0': 'Location_Name'},
+    'huc_8_code': {'2.2': 'HUCEightDigitCode', '3.0': 'Location_HUCEightDigitCode'},
+    'provider': {'2.2': 'ProviderName', '3.0': 'ProviderName'},
+    'phys_chem_results': {'2.2': 'resultPhysChem', '3.0': 'fullPhysChem'},
+    'observed_property': {'2.2': 'CharacteristicName', '3.0': 'Result_Characteristic'},
+    'sampling_media': {'2.2': 'ActivityMediaName', '3.0': 'Activity_Media'},
+    'sample_fraction': {'2.2': 'ResultSampleFractionText', '3.0': 'Result_SampleFraction'},
+    'depth_height': {'2.2': 'ResultDepthHeightMeasure/MeasureValue', '3.0': 'ResultDepthHeight_Measure'},
+    'aggregation_duration': {'2.2': 'ResultTimeBasisText', '3.0': 'Result_TimeBasis'},
+    'sampling_temp': {'2.2': 'ResultTemperatureBasisText', '3.0': 'Result_MeasureTemperatureBasis'},
+    'statistic': {'2.2': 'StatisticalBaseCode', '3.0': 'Result_StatisticalBase'},
+    'start_date': {'2.2': 'ActivityStartDate', '3.0': 'Activity_StartDate'},
+    'start_time': {'2.2': 'ActivityStartTime/Time', '3.0': 'Activity_StartTime'},
+    'start_time_zone': {'2.2': 'ActivityStartTime/TimeZoneCode', '3.0': 'Activity_StartTimeZone'},
+    'end_date': {'2.2': 'ActivityEndDate', '3.0': 'Activity_EndDate'},
+    'end_time': {'2.2': 'ActivityEndTime/Time', '3.0': 'Activity_EndTime'},
+    'end_time_zone': {'2.2': 'ActivityEndTime/TimeZoneCode', '3.0': 'Activity_EndTimeZone'},
+    'result': {'2.2': 'ResultMeasureValue', '3.0': 'Result_Measure'},
+    'result_unit': {'2.2': 'ResultMeasure/MeasureUnitCode', '3.0': 'Result_MeasureUnit'},
+    'result_status': {'2.2': 'ResultStatusIdentifier', '3.0': 'Result_MeasureStatusIdentifier'},
+    'result_type': {'2.2': 'ResultValueTypeName', '3.0': 'Result_MeasureType'},
+    'activity_type': {'2.2': 'ActivityTypeCode', '3.0': 'Activity_TypeCode'},
+    'depth_height_unit': {'2.2': 'ResultDepthHeightMeasure/MeasureUnitCode', '3.0': 'ResultDepthHeight_MeasureUnit'},
+    'depth_height_ref': {'2.2': 'ResultDepthAltitudeReferencePointText', '3.0': 'ResultDepthHeight_AltitudeReferencePoint'}
+}
+
+EPA_GEOSERVER_WFS_TIMEOUT_LIMIT = os.environ.get('EPA_GEOSERVER_WFS_TIMEOUT_LIMIT', 0.5)
 
-def _post_wqp_data_search(query_data: dict) -> Optional[requests.Response]:
+
+def _post_wqp_search(search_type: str, query_data: dict,
+                     api_version: str = EPA_WQP_API_VERSION) -> Optional[requests.Response]:
     """
     Request the locations information
+    :param search_type:
     :param query_data:
-    :return:
+    :param api_version:
```

Comment:
provide descriptions of the parameters.  What are valid search types?  I assume that it is a controlled list.  Same for the api version.